Makefile

```diff
@@ -20,9 +20,13 @@ license:
 	@scripts/check_license.sh
 
 .PHONY: unit-test
-unit-test:
+unit-test: generate-test-keys
 	@scripts/check_unit.sh
 
+generate-test-keys: clean
```

Comment:
I think it's better to have artifacts for unit tests be generated with the go generate conventions.

Reply:
(also - it feels a bit odd to regenerate every time we do make unit-test).

Reply:
will try to incorporate `//go:generate openssl....` in the test code

Reply:
done in a separate PR and will be merged right after this PR